### PR TITLE
WIP Can introspect the whole project in a single batch.

### PR DESCRIPTION
**Pull request by jpakkane**
Let's get the Qt Creator et al integration rolling.

### CODE REVIEW

File: mesonbuild/mintro.py [hunk 1]
```diff
@@ -29,6 +29,8 @@
 def add_arguments(parser):
     parser.add_argument('--targets', action='store_true', dest='list_targets', default=False,
                         help='List top level targets.')
+    parser.add_argument('--target-batch', action='store_true', dest='target_batch', default=False,
+                        help='Batch export info on all targets.')
```
Comment (jon-turney) [inline mesonbuild/mintro.py]: Would '--targets-all' or similar be a better name for this, since that's what it does?

Comment (jpakkane) [reply mesonbuild/mintro.py]: It is meant to export everything about a target needed by an IDE, not just files. We'd need to get some feedback from IDE devs. (For some reason github prevents me from tagging TH of Qt Creator on this post.)

**Comment by textshell** [comment]
I think this is mostly superseded by #4547. Which which is the preferred approach of the developers working on the kdevelop and qtcreator integration.

**Comment by jpakkane** [comment]
Off to the void with this then.

**Pull request closed without merging**